SeaPublicWebsite/DataStores/PropertyDataStore.cs

```diff
@@ -80,10 +80,12 @@ public async Task<PropertyData> CreateNewPropertyDataAsync()
                 Reference = RandomHelper.Generate8CharacterReference()
             };
             attemptedReferences.Add(propertyData.Reference);
-            
+
             try
             {
                 await dataAccessProvider.AddPropertyDataAsync(propertyData);
+                propertyData.RecommendationsFirstRetrievedAt = DateTime.Now.ToUniversalTime();
+                await SavePropertyDataAsync(propertyData);
```

Comment:
```suggestion
                propertyData.RecommendationsFirstRetrievedAt = DateTime.Now.ToUniversalTime();
                await dataAccessProvider.AddPropertyDataAsync(propertyData);
```
Currently saving it, updating it, and then immediately saving it again.